Bottom: 4cc845f5a9ef902a5648165877e570eac8aa5174
Top:    678b7ad4c1bb36b1f5f7d6ccaa2c32cff930e92d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-23 15:32:47 +0000

Disable DRC checks on "outline" and "route" layers


---

diff --git a/src/find.c b/src/find.c
index 4edb048..2f6dd76 100644
--- a/src/find.c
+++ b/src/find.c
@@ -728,11 +728,10 @@ LookupLOConnectionsToPVList (int flag, bool AndRats)
         {
           LayerType *layer = LAYER_PTR (layer_no);
 
-          if (layer->no_drc)
-             continue;
+          if (layer->no_drc || is_layer_mechanical (layer))
+            continue;
 
           info.layer = layer_no;
-
           /* add touching lines */
           if (setjmp (info.env) == 0)
             r_search (layer->line_tree, &search_box,
@@ -3635,6 +3634,9 @@ DRCAll (void)
     {
       COPPERLINE_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
+
         /* check line clearances in polygons */
         if (PlowsPolygon (PCB->Data, LINE_TYPE, layer, line, drc_callback, &info))
           {
@@ -3680,6 +3682,8 @@ DRCAll (void)
     {
       COPPERARC_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
         if (PlowsPolygon (PCB->Data, ARC_TYPE, layer, arc, drc_callback, &info))
           {
             IsBad = true;
diff --git a/src/misc.c b/src/misc.c
index 040124c..448ec45 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2453,6 +2453,14 @@ ElementOrientation (ElementType *e)
   return dy > 0 ? 3 : 1;
 }
 
+bool is_layer_mechanical (LayerType *layer)
+{
+  if (strcmp (layer->Name, "outline") == 0 ||
+      strcmp (layer->Name, "route") == 0)
+    return true;
+  return false;
+}
+
 int
 ActionListRotations(int argc, char **argv, Coord x, Coord y)
 {
diff --git a/src/misc.h b/src/misc.h
index 85178db..004708e 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -139,6 +139,8 @@ char * GetInfoString (void);
    comparing two similar footprints.  */
 int ElementOrientation (ElementType *e);
 
+bool is_layer_mechanical (LayerType *layer);
+
 /* These are in netlist.c */
 
 void NetlistChanged (int force_unfreeze);
